server: de-flake TestClusterAPI

tbg opened the pull request:
The test was overriding a settings value, but with the right interleaving
with a SQL migration that also updated it, the test would get stuck
with a `true` when it wanted a `false`. Repeatedly override the settings
value until the right result comes out instead.

Fixes #19934.
Fixes #19935.
Fixes #19936.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/21029)
<!-- Reviewable:end -->

nvanbenschoten commented on pkg/server/admin_test.go:
```diff
@@ -1060,6 +1059,10 @@ func TestClusterAPI(t *testing.T) {
 			// We need to retry, because the cluster ID isn't set until after
 			// bootstrapping.
 			testutils.SucceedsSoon(t, func() error {
+				// In particular, we need to repeatedly set this value, as there is
```
Would it be possible to detect when the SQL migration is complete and just wait for that right after starting the server?

tschottdorf replied:
I've switched to using `SET CLUSTER SETTING` which doesn't have this problem in the first place. Thanks for giving me that *brilliant* idea 😄

tbg committed:
server: de-flake TestClusterAPI

The test was overriding a settings value, but with the right interleaving with a
SQL migration that also updated it, the test would get stuck with a `true` when
it wanted a `false`. Setting the option via the proper SQL command avoids this
issue because that properly serializes these updates.

Fixes #19934.
Fixes #19935.
Fixes #19936.

Release note: None
